src/Compilers/CSharp/Portable/Lowering/AsyncRewriter/RuntimeAsyncRewriter.cs

```diff
@@ -70,6 +70,28 @@ public static BoundStatement Rewrite(
         }
     }
 
+    public static BoundStatement RewriteWithoutHoisting(
```

Comment:
Consider adding documentation for this method, and asserting the shape of the method symbol coming in so we only call this in specific scenarios.